cpu/atxmega/periph/spi.c

```diff
@@ -87,6 +130,36 @@ void spi_init_pins(spi_t bus)
     gpio_init(spi_config[bus].mosi_pin, GPIO_OUT);
 }
 
+uint32_t spi_clk_info(spi_t bus, spi_clk_t clk)
+{
+    (void)bus;
+    (void)clk;
+    /* in spi_acquire :
+     * CLK2X    PRESCALER
+     * 1        01          ->  ClkPER / 8
+     * All clock prescalers was set to 1 so ClkPER = CLOCK_CORECLOCK */
+    return CLOCK_CORECLOCK / 8;
+}
+
+spi_clk_t spi_get_clk(spi_t bus, uint32_t freq)
+{
+    (void)bus;
+    /* bound divider from 2 to 128 */
+    if (freq > CLOCK_CORECLOCK / 2) {
+        freq = CLOCK_CORECLOCK / 2;
+    }
+    assert(freq >= CLOCK_CORECLOCK / 128);
+
+    return _clk_shift(freq);
```

Comment:
```suggestion
    uint8_t shift = _clk_shift(freq);
    return ((~shift & 1) << 7) | (shift >> 1)
```

Maybe prepare the value directly in the format that it can be or'ed into the CTRL register?